content/doc/developer/security/xss-prevention.adoc

```diff
@@ -96,3 +96,65 @@ To pass arguments to localized expressions without escaping them in escaped-by-d
 
 [WARNING]
 If the source of the argument to `j:out` or `Functions#rawHtml` isn't completely trusted, these might result in an XSS vulnerability.
+
+=== Passing values to JavaScript
+
+==== The bad way
+
+In a situation where you need to pass information to the JavaScript, you could be inclined to inject the value directly inside the script like in the following snippet. 
+
+[source, xml]
+----
+<?jelly escape-by-default='true'?>
+<j:jelly xmlns:j="jelly:core">
+  <h1>Random title</h1>
+  <div id="target-div"></div>
+
+  <script>
+     var textToInsert = "${variableFromJava}";
+     document.querySelector('#target-div').innerText = textToInsert;
+  </script>
+</j:jelly>
+----
+
+In this case, if the `variableFromJava` comes from unsafe source, it could be used to create an XSS. 
```

Comment:
```suggestion
In this case, if the `variableFromJava` comes from an unsafe source, it could be used to create a cross-site scripting (XSS) attack.
```